### ISSUE TITLE
Module name clashes

### ISSUE THREAD

**Issue by meyer1994** [aws/serverless-application-model#1911]
### Description:

When using this library, or any library that uses it, there are name clashes
with common names used in projects. Like `utils`. This is the result of a
random `sys.path.insert` at `samtranslator/feature_toggle/feature_toggle.py`.

This was not breaking before, I do not know on which version it started. This
breaks CI with unpinned versions. Or, for example, it breaks when any
dependency uses this module. For example [moto][1]:

```
moto > cfn-lint > aws-sam-translator
```

Either way, modules should not modify `sys.path` willy-nilly.

### Steps to reproduce:

1. Create an `utils.py`
1. Create a `main.py` file that first imports something that reaches
`feature_toggle.py`
1. Try importing something from your `utils.py`

#### Example

```py
# utils.py
something = 'nice'

# main.py
from samtranslator.translator.translator import Translator
from utils import something  # ImportError
```

### Observed result:

`ImportError` is thrown when trying to import things from my `utils.py`.

### Expected result:

No `ImportError`.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Any
2. If using SAM CLI, `sam --version`:
3. AWS region: Does not matter

`Add --debug flag to any SAM CLI commands you are running`

[1]: https://github.com/spulec/moto

**Comment by wyattanderson** [comment]
Also running into this. Looks like it was introduced in #1737.

**Comment by jfuss** [comment]
This was patched in [v1.46.0](https://github.com/aws/serverless-application-model/releases/tag/v1.46.0). Closing
